### ISSUE TITLE
admin/general_settings/edit is missing form validation error messages and fails to save.

### ISSUE THREAD

**Issue by stabenfeldt** [solidusio/solidus#800]
Hi,

I'm running 1.3.0.alpha
If I only add something to title and save, then I get a *General Settings has been successfully updated!* flash message. But nothing is saved.

I threw in a byebug in `admin/general_settings_controller.rb`
```ruby
 11       def update
 12         byebug
 13         params.each do |name, value|
 14           next unless Spree::Config.has_preference? name
 15           Spree::Config[name] = value
 16         end
 17
 18         current_store.update_attributes store_params
 19
 20         flash[:success] = Spree.t(:successfully_updated, resource: Spree.t(:general_settings))
 21         redirect_to edit_admin_general_settings_path
 22       end
```

```ruby
(byebug)  current_store
#<Spree::Store id: nil, name: "test", url: "test.com", meta_description: "test", meta_keywords: "test", seo_title: "test", mail_from_address: "test@test.com", default_currency: nil, code: nil, default: false, created_at: nil, updated_at: nil>
(byebug)  current_store.save!
   (0.4ms)  BEGIN
   (0.2ms)  ROLLBACK
*** ActiveRecord::RecordInvalid Exception: Validation failed: Code can't be blank

```

It looks like it's missing *Code*.
There is no fields named code on the general settings page:
![screen shot 2016-02-04 at 20 25 42](https://cloud.githubusercontent.com/assets/181272/12826895/7f8acf82-cb7d-11e5-890e-0d348f3709e1.png)

What is the missing code? 

Cheers,
Martin

**Comment by jhawthorn** [comment]
Thanks for the report

Clearly we're ignoring the result of `update_attributes`, and always setting successfully updated, which we should not do. This is because this configuration page has changed a lot over the years. This used to just set preferences (which couldn't really fail), but now sets settings on a `Store` model.

We'll want to add something to the extent of

``` ruby
@store = current_store
if @store.update_attributes(store_params)
  # success
else
  # failed
end
```

But it could need some larger changes to properly use the `@store` variable and display errors.

**Comment by jhawthorn** [comment]
@stabenfeldt this should be fixed by #819. Thanks again.

**Comment by stabenfeldt** [comment]
@jhawthorn Awesome, thanks for fixing the issue so quickly! :-)
